services/workflows-service/src/transaction/dtos/transaction-create.dto.ts

```diff
@@ -33,10 +34,14 @@ export class CounterpartyInfo {
   @ApiProperty({ required: false }) @IsString() @IsOptional() sortCode?: string;
   @ApiProperty({ required: false }) @IsString() @IsOptional() bankCountry?: string;
 
-  @IsOptional()
+  @ApiProperty({ type: OmitType(BusinessCreateDto, ['correlationId']) })
+  @ValidateIf(obj => !obj.endUserData)
+  @ValidateNested()
   businessData?: Omit<BusinessCreateDto, 'correlationId'>;
 
-  @IsOptional()
+  @ApiProperty({ type: OmitType(EndUserCreateDto, ['correlationId']) })
+  @ValidateIf(obj => !obj.businessData)
+  @ValidateNested()
   endUserData?: Omit<EndUserCreateDto, 'correlationId'>;
```

Comment:
Can be implemented by ValidatorConstraintInterface which allow at least one field to be present
https://github.com/typestack/class-validator/issues/1581



```
// at-least-one.decorator.ts
import { ValidationArguments, ValidatorConstraint, ValidatorConstraintInterface } from 'class-validator';

@ValidatorConstraint({ name: 'atLeastOne', async: false })
export class AtLeastOneConstraint implements ValidatorConstraintInterface {
    validate(value: any, args: ValidationArguments) {
        const { object } = args;
        const correlatedFields = args.constraints[0];
        return correlatedFields.some(field => object[field] !== undefined);
    }

    defaultMessage(args: ValidationArguments) {
        return `At least one of the fields ${args.constraints[0].join(', ')} must be present`;
    }
}
```

Reply:
```
import { AtLeastOneConstraint } from './at-least-one.decorator';

export class MyDTO {
    @IsString()
    propertyA?: string;

    @IsString()
    propertyB?: string;

    @Validate(AtLeastOneConstraint, ['propertyA', 'propertyB'])
    atLeastOnePresent: string;
}
```